docs/documentation.html

```diff
@@ -833,6 +833,41 @@ <h3 id="grammar-syntax-and-semantics-parsing-expression-types">Parsing Expressio
     </div>
   </dd>
 
+  <dt><code>!.</code> (END OF INPUT)</dt>
+
+  <dd>
+    <p>Match END OF INPUT. This <em>Bang Dot</em> sequence will match a completely empty input,
+      or patterns up to the end of input. This is a composite rule;
```

Comment:
"patterns up to the end of the rule" isn't clear.  I think you mean that it's useful when used as a predicate, but we probably need some more words.